eg-api.php

```diff
@@ -188,6 +195,11 @@
 		error_log("emailPetitions was not set");
 	}
 	error_log("Finished api request.");
+	if (validAPIKey($_REQUEST, $db)) {
+		writeToResourceLog(validAPIKey($_REQUEST, $db),"eg-api.php",$log_trail);
+	} else {
+		writeToResourceLog("unknown user","eg-api.php",$log_trail);
```

Comment:
This seems problematic.  The DB has an smallint for the userid, but this line is sending a string.

Reply:
Good catch. I hadn't thought about userid being an int. 

What about -1 or 0 for there, instead?